packages/backend/src/database/entities/savedQueries.ts

```diff
@@ -317,6 +387,53 @@ export const createSavedQueryVersion = async (
     }: CreateSavedQueryVersion,
 ): Promise<void> => {
     await db.transaction(async (trx) => {
+        const pivotDimensions = chartConfig.seriesLayout.groupDimension
+            ? [chartConfig.seriesLayout.groupDimension]
+            : undefined;
+        let convertedChartType: DbSavedQueryVersion['chart_type'] = 'cartesian';
+        let convertedChartConfig:
+            | DbSavedQueryVersion['chart_config']
+            | undefined;
+        switch (chartConfig.chartType) {
+            case DBChartTypes.BIG_NUMBER:
+                convertedChartType = 'big_number';
+                convertedChartConfig = {};
+                break;
+            case DBChartTypes.TABLE:
+                convertedChartType = 'table';
+                convertedChartConfig = '';
```

Comment:
Should this be an empty object ? not sure how typescript is not complaining about this 🤔

Reply:
Or should this and big number set the chartconfig to undefined ?

Reply:
Good catch!

No typescript errors, which is odd. This actually happened earlier further down in the code when I construct the `CartesianChartConfig` the `Series` type had a wrong field but still no complaints from typescript. It was the same variable `convertedChartConfig`. Is there anything I can do to harden the typing?

Reply:
I get this in my IDE but can still assign `''` and an object not matching a valid chart config

<img width="509" alt="Screenshot 2022-02-24 at 10 03 34" src="https://user-images.githubusercontent.com/11660098/155502757-6e5bddcb-6b76-4fe6-baaf-082bcd5378b9.png">

Reply:
The empty string, I have no idea why 🤔 
The empty object is because you defined it has an empty object here:
```
type BigNumberConfig = {
    type: 'big_number';
    config: {};
};

type TableChartConfig = {
    type: 'table';
    config: {};
};
```

Reply:
Also, `DbSavedQueryVersion` doesn't enforce the config_chart to match the config_type. 
So we could have type Big_Number and
```
{
xField: ...
yField: ...
flipAxes: ...
}
```

Reply:
Will fix in the next refactor of the backend